pkg/loki/modules.go

```diff
@@ -1158,6 +1158,15 @@ func (t *Loki) addCompactorMiddleware(h http.HandlerFunc) http.Handler {
 func (t *Loki) initIndexGateway() (services.Service, error) {
 	t.Cfg.IndexGateway.Ring.ListenPort = t.Cfg.Server.GRPCListenPort
 
+	var shardingStrategy indexgateway.ShardingStrategy
+	if t.Cfg.IndexGateway.Mode != indexgateway.RingMode || t.indexGatewayRingManager.Mode == indexgateway.ClientMode {
+		shardingStrategy = indexgateway.NewMatchAllStrategy()
+	} else {
+		instanceAddr := t.indexGatewayRingManager.RingLifecycler.GetInstanceAddr()
+		instanceID := t.indexGatewayRingManager.RingLifecycler.GetInstanceID()
+		shardingStrategy = indexgateway.NewShuffleShardingStrategy(t.indexGatewayRingManager.Ring, t.Overrides, instanceAddr, instanceID)
```

Comment:
nitpicking: WDYT of just passing `t.indexGatewayRingManager` to `NewShuffleShardingStrategy` and extracting the desired attrs there?  otherwise for every new attr that you might be interested you'd be adding a new parameter to this method

Reply:
Hm, not sure if a ShardingStrategy should necessarily know about a RingManager. Maybe a new constructor `GetShardingStrategyForRing()`